### PR TITLE
Bug: Unable to delete autocomplete history items

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/488551667048375/1208592488628219/f

### Description
Fixed item deletion from suggestions.

### Steps to test this PR

Test in Browser:
- [x] Install from this  branch
- [x] Search for something, i.e. "dog food".
- [x] Search for "dog".
- [x] Long press and delete the "dog food" entry.
Verify:
- [x] That the keyboard is shown.
- [x] That the autocomplete list is shifted up (So the position of the entry is visible).
- [x] That the entry is deleted from the list.

Test using the widget
- [x] Install from this branch
- [x] Add the widget.
- [x] Search for something from the widget, i.e. "dog food".
- [x] Search for "dog".
- [x] Long press and delete the "dog food" entry.
Verify:
- [x] That the keyboard is shown.
- [x] That the autocomplete list is shifted up (So the position of the entry is visible).
- [x] That the entry is deleted from the list.

### CODE REVIEW

**Comment by anikiki** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5168?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5168** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5168?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5165** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5165?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5123** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5123?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5105** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5105?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5065** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5065?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5064** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5064?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5063** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5063?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5042** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5042?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5019** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5019?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5018** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/tabs/db/TabsDao.kt [hunk 1]
```diff
@@ -45,10 +44,6 @@ abstract class TabsDao {
     @Query("select * from tabs where deletable is 0 order by position")
     abstract fun flowTabs(): Flow<List<TabEntity>>
 
-    // TODO: ANA to remove this
-    @Query("select * from tabs where deletable is 0 order by position")
-    abstract fun singleTabs(): Single<List<TabEntity>>
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/tabs/db/TabsDao.kt:50]: Removed also some unused code.

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt [hunk 2]
```diff
@@ -82,8 +81,6 @@ class TabDataRepository @Inject constructor(
 
     private var purgeDeletableTabsJob = ConflatedJob()
 
-    override fun getTabsObservable(): Single<List<TabEntity>> = tabsDao.singleTabs()
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt:85]: Same, removed unused code.

File: browser-api/build.gradle [hunk 3]
```diff
@@ -35,10 +35,6 @@ dependencies {
 
     // LiveData
     implementation AndroidX.lifecycle.liveDataKtx
-
-    // TODO: ANA to remove this
-    implementation "io.reactivex.rxjava2:rxjava:_"
-    implementation "io.reactivex.rxjava2:rxandroid:_"
```
Comment (anikiki) [inline browser-api/build.gradle:41]: Removed unneeded dependencies.

File: browser-api/src/main/java/com/duckduckgo/app/tabs/model/TabRepository.kt [hunk 4]
```diff
@@ -44,8 +43,6 @@ interface TabRepository {
 
     val tabSwitcherData: Flow<TabSwitcherData>
 
-    fun getTabsObservable(): Single<List<TabEntity>>
```
Comment (anikiki) [inline browser-api/src/main/java/com/duckduckgo/app/tabs/model/TabRepository.kt:47]: Unused.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 5]
```diff
@@ -3696,7 +3696,7 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
-    fun autoCompleteSuggestionsShown() {
+    fun onAutoCompleteSuggestionsChanged() {
```
Comment (anikiki) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:3699]: Renamed this.

**Review by joshliebe — APPROVED** [review]
Nice! Tested, works as expected now 🙂

**Pull request merged**